### PR TITLE
WIP on time sources

### PR DESCRIPTION
**Pull request by adamdbrw**
Signed-off-by: Adam Dabrowski <adam.dabrowski@robotec.ai>

### CODE REVIEW

**Comment by pijaro** [comment]
I think it would be the best to add some kind of `TimeSourceFactory` and let users choose which time source to use while creating a node.

File: src/Ros2ForUnity/Scripts/Time/ROS2Clock.cs [hunk 1]
```diff
@@ -19,57 +19,46 @@ namespace ROS2
...
-    private int seconds;
-    private uint nanoseconds;
-    private ROS2.Clock clock;
+    private ITimeSource _timeSource;
 
-    public ROS2Clock()
-    {
-        clock = new ROS2.Clock();
+    public ROS2Clock() : this(new DotnetTimeSource())
+    {   // By default, use DotnetTimeSource
```
Comment (pijaro) [inline src/Ros2ForUnity/Scripts/Time/ROS2Clock.cs:29]: Why though? Shouldn't ros2 be the default?

Comment (adamdbrw) [reply src/Ros2ForUnity/Scripts/Time/ROS2Clock.cs]: I also changed this to ros2 after some consideration

**Comment by adamdbrw** [comment]
@pijaro review the new version now - candidate to merge.

**Comment by pijaro** [comment]
lgtm

**Pull request merged**